template_funcs/funcmap.go

```diff
@@ -58,9 +58,9 @@ var FuncMap = template.FuncMap{
 	"quoteMeta":   regexp.QuoteMeta,
 
 	// Filepath manipulation
-	"base":     filepath.Base,
-	"clean":    filepath.Clean,
-	"dir":      filepath.Dir,
+	"base":     func(path string) string { return filepath.ToSlash(filepath.Base(path)) },
+	"clean":    func(path string) string { return filepath.ToSlash(filepath.Clean(path)) },
+	"dir":      func(path string) string { return filepath.ToSlash(filepath.Dir(path)) },
```

Comment:
Is this correct? Why would you want this? If I'm reading correctly this will replace all `\` characters in the Windows path with `/`.

Reply:
Interesting, so the tests don't pass unless you use `filepath.ToSlash`? I would be curious to know why. I am just suspicious of this change because someone might, for example, want to read the contents of a file (copyright text, for example), the path of which might need to pass through one of these functions. I don't know if Windows would understand you trying to open a file with `/` characters.

Kind of sad that the Go Playground does not seem to have Windows environments I can test this out with.

Reply:
The unittests in `funcmap_test.go` for these template functions expect the result to contain slashes (`/home/file.txt`). Without the change, you get `\home\file.txt` as result, from the `clean` template function test. So you would have to have OS-specific unittests.
Believe you me, nobody wants this.
And go APIs on Windows work just fine with slashes (C APIs too for that matter). Only when you build actual commandlines for consumption by `cmd.exe` do you need backslashes.

Reply:
BTW: everywhere pathes are now stored in the coding in my PR, they are stored after conversion to slashes. As mentioned, opening files etc. works just fine with these on Windows.